src/index.js

```diff
@@ -6,83 +6,202 @@ const { ApolloServer, gql } = require('apollo-server');
   },
   {
     id: 3,
-    title: 'The Cave',
+    name: 'The Cave',
     author: 3,
+    tags: ['Adventure'],
   },
 ];
 
 const users = [
   {
     id: 1,
-    name: 'J.K. Rowling',
+    firstName: 'J.K.',
+    lastName: 'Rowling',
     address: 'U.K.',
-    favoriteBook: 1,
+    dob: '21/03/1980',
+    interests: ['SciFi', 'Adventure'],
+    isActive: true,
+    favoriteBooks: [1],
   },
   {
     id: 2,
-    name: 'Michael Crichton',
+    firstName: 'Michael',
+    lastName: 'Crichton',
     address: 'USA',
-    favoriteBook: 2,
+    dob: '04/03/1970',
+    interests: ['SciFi'],
+    isActive: true,
+    favoriteBooks: [1, 2],
   },
   {
     id: 3,
-    name: 'Batman',
+    firstName: 'Batman',
+    lastName: '',
     address: 'Gotham',
-    favoriteBook: 1,
+    dob: '21/03/1988',
+    interests: ['Adventure'],
+    isActive: true,
+    favoriteBooks: [1, 3],
   },
   {
     id: 4,
-    name: 'Spiderman',
+    firstName: 'Spiderman',
+    lastName: 'Amazing',
     address: 'New York',
-    favoriteBook: 2,
+    dob: '23/07/1992',
+    interests: ['SciFi', 'Adventure'],
+    isActive: false,
+    favoriteBooks: [1, 2, 3],
+  },
+];
+
+const publications = [
+  {
+    id: 1,
+    name: 'Rolling',
+    author: 'J.K. Rolling',
+    tags: ['SciFi', 'Adventure'],
+  },
+];
+
+const CopyStatus = [
+  {
+    id: 1,
+    name: 'Rolling',
+    author: 'J.K. Rolling',
+    tags: ['SciFi', 'Adventure'],
+    book: 1,
+    borrower: 1,
+    status: 'Checked',
+    quality: 'Excellent',
+  },
+  {
+    id: 2,
+    name: 'Huff',
+    author: 'Loof',
+    tags: ['Adventure'],
+    book: 2,
+    borrower: 1,
+    status: 'Available',
+    quality: 'Good',
+  },
+  {
+    id: 1,
+    name: 'Rolling',
+    author: 'J.K. Rolling',
+    tags: ['SciFi', 'Adventure'],
+    book: 1,
+    borrower: 1,
+    status: 'Checked',
+    quality: 'Excellent',
   },
 ];
 
 // Type definitions define the "shape" of your data and specify
 // which ways the data can be fetched from the GraphQL server.
 const typeDefs = gql`
   # Comments in GraphQL are defined with the hash (#) symbol.
-
   # This "Book" type can be used in other type declarations.
-  type Book {
-    title: String
+
+  interface Publication {
+    id: ID!
+    name: String
     author: User
+    image: String
+    tags: [Tag]
   }
 
-  type User {
+  type Book implements Publication {
+    id: ID!
     name: String
+    author: User
+    image: String
+    tags: [Tag]
+  }
+
+  type CopyStatus implements Publication {
+    id: ID!
+    name: String
+    author: User
+    image: String
+    tags: [Tag]
+    book: Book
+    borrower: User
+    status: BookStatus
+    quality: Quality
+  }
+
+  type User {
+    id: ID!
+    firstName: String
+    lastName: String
     address: String
-    favoriteBook: Book
+    photo: String
+    dob: String
+    interests: [Tag]
+    isActive: Boolean
+    favoriteBooks: [Book]
+  }
+
+  enum Quality {
+    Poor
+    Good
+    Excellent
+  }
+
+  enum BookStatus {
```

Comment:
This should be called `CopyStatus`. A `Copy` of a book is what the users will actually check out. A `Copy` can go missing, but not the book.